lib/curly/presenter.rb

```diff
@@ -47,13 +47,15 @@ def initialize(context, options = {})
 
       self.class.presented_names.each do |name|
         value = options.fetch(name) do
-          default_values.fetch(name) do
+          default = default_values.fetch(name) do
             raise ArgumentError.new("required identifier `#{name}` missing")
           end
-        end
 
-        if value.respond_to?(:call)
-          value = value.call
+          if default.respond_to?(:call)
+            instance_exec(name, &default)
```

Comment:
That's true, but it will automatically call the `call` method on any default value that has it, which seems bad to me. I think the Ruby Way(tm) is to provide a good old block when calling the `presents` method.